packages/message/src/Queue.ts

```diff
@@ -40,46 +45,66 @@ export const getQueueNameByUrl = (
  * @alias module:Queue
  */
 export const getQueueName = (message: Message.CumulusMessage) => {
+  util.deprecate('@cumulus/message/Queue.getQueueName', '1.24.0');
   const queueName = get(message, 'cumulus_meta.queueName');
   if (isNil(queueName)) {
     throw new Error('cumulus_meta.queueName not set in message');
   }
   return queueName;
 };
 
+/**
+ * Get the queue URL from a workflow message.
+ *
+ * @param {CumulusQueueMessage} message - A workflow message object
+ * @returns {string} A queue URL
+ * @throws {Error} if no queue URL in the message
+ *
+ * @alias module:Queue
+ */
+export const getQueueUrl = (message: CumulusQueueMessage): string => {
+  const queueUrl = message.cumulus_meta?.queueUrl;
```

Comment:
If this is a `CumulusQueueMessage`, that seems to imply that `message.cumulus_meta.queueUrl` should always be present? You might want to extend `CumulusMessage` in this case to indicate that.